### PR TITLE
Reduce memory used by inodes in the context

### PR DESCRIPTION
**Pull request by sebastiencs**

### CODE REVIEW

File: tezos/context/src/working_tree/storage.rs [hunk 1]
```diff
@@ -239,6 +271,19 @@ impl From<DirEntryIdError> for StorageError {
...
 
+impl From<FatPointerIdError> for StorageError {
+    fn from(_: FatPointerIdError) -> Self {
+        Self::FatPointerIdError
+    }
+}
+
+impl From<std::convert::Infallible> for StorageError {
+    fn from(_: std::convert::Infallible) -> Self {
+        // This implementation exists only to be able to use `?` on a Result<_, Infallible>
```
Comment (tizoc) [inline tezos/context/src/working_tree/storage.rs:282]: cool, I didn't know about `Infallible`

File: tezos/context/src/working_tree/storage.rs [hunk 2]
```diff
@@ -399,175 +444,478 @@ impl TryFrom<usize> for InodeId {
...
+impl From<InodeId> for u32 {
+    fn from(v: InodeId) -> Self {
+        v.0
+    }
+}
+
+#[derive(Clone, Debug, Copy, PartialEq, Eq)]
+pub struct FatPointerId(u32);
+
+impl TryInto<usize> for FatPointerId {
```
Comment (tizoc) [inline tezos/context/src/working_tree/storage.rs:462]: @sebastiencs when can this fail?

**Review by tizoc — APPROVED** [review]
@sebastiencs look ok to me, but to be honest, I only looked at it very superficially (it is quite a big diff). What I would like is more documentation on the new types that have been defined, while I understand the overall approach, I am not very clear on the specific components.

If it is well tested already, feel free to merge it, otherwise I can give it another pass after you have added some more docs.

File: tezos/context/src/working_tree/storage.rs (hunk 2 above)
Comment (sebastiencs) [reply tezos/context/src/working_tree/storage.rs:462]: Ah yes, it cannot fail. I will change the trait to `Into`

**Comment by sebastiencs** [comment]
@tizoc Thanks, I will add more documentation

**Comment by sebastiencs** [comment]
@tizoc I've updated the PR with more documentations

**Comment by tizoc** [comment]
Weird, I don't know why the comments I added are not associated with the lines. But doesn't matter, those were not important.

**Review by tizoc — APPROVED** [review]

**Pull request merged**